File: Algoritimo-Estrutruras-de-Dados/Aeds-ll/2025_01/exercicios/lab_01/questao_02/AquecimentoRec.java
import java.util.*;

public class AquecimentoRec{
    public static int IsFim(String str){
        int result = 1;

        if(str.length() == 3 && str.charAt(0) == 'F' && str.charAt(1) == 'I' && str.charAt(2) == 'M'){
            result = 0;
        }

        return result;
    }

    public static int ContarRec(String str, int index){
        int count = 0;

            if(index < str.length()){
                if(str.charAt(index) >= 'A' && str.charAt(index) <= 'Z'){
                    count++;
                }
                count = count + ContarRec(str, index + 1);
            }

        return count;
    }

    public static int Contar(String str){
        return ContarRec(str, 0);
    }

    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        String line;

        line = sc.nextLine();
        while(IsFim(line) != 0){
            System.out.println(Contar(line) + "");

            line = sc.nextLine();
        }
        sc.close();
    }
}

File: Algoritimo-Estrutruras-de-Dados/Aeds-ll/2025_01/exercicios/lab_01/questao_04/AquecimentoRec.c
#include <stdio.h>
#include <string.h>

int IsFim(char str[]){
    int result = 1;

    if(str[0] == 'F' && str[1] == 'I' && str[2] == 'M'){
        result = 0;
    }

    return result;
}

int ContarRec(char str[], int index){
    int count = 0;
    int n = strlen(str);

    if(index < n){
        if(str[index] >= 'A' && str[index] <= 'Z'){
            count++;
        }
        count = count + ContarRec(str, index + 1);
    }

    return count;
}

int Contar(char str[]){
    return ContarRec(str, 0);
}

int main(void){
    char line[100];

    scanf("%[^\n]", line);
    while(IsFim(line) != 0){
        printf("%d\n", Contar(line));

        scanf(" %[^\n]", line);
    }

  return 0;
}